lib/components/wallet_connect/pair.dart

```diff
@@ -0,0 +1,217 @@
+// ignore_for_file: prefer_const_constructors
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:qubic_wallet/di.dart';
+import 'package:qubic_wallet/flutter_flow/theme_paddings.dart';
+import 'package:qubic_wallet/l10n/l10n.dart';
+import 'package:qubic_wallet/services/wallet_connect_service.dart';
+import 'package:qubic_wallet/stores/application_store.dart';
+import 'package:qubic_wallet/styles/edge_insets.dart';
+import 'package:qubic_wallet/styles/text_styles.dart';
+import 'package:qubic_wallet/styles/themed_controls.dart';
+import 'package:walletconnect_flutter_v2/walletconnect_flutter_v2.dart';
+
+class Pair extends StatefulWidget {
+  final PairingMetadata? pairingMetadata;
+  final List<String> pairingMethods;
+  final List<String> pairingEvents;
+  final int pairingId;
+  final Map<String, Namespace>? pairingNamespaces;
+
+  const Pair(
+      {super.key,
+      required this.pairingId,
+      required this.pairingMethods,
+      required this.pairingMetadata,
+      required this.pairingEvents,
+      required this.pairingNamespaces});
+
+  @override
+  // ignore: library_private_types_in_public_api
+  _PairState createState() => _PairState();
+}
+
+class _PairState extends State<Pair> {
+  final ApplicationStore appStore = getIt<ApplicationStore>();
+  final WalletConnectService wcService = getIt<WalletConnectService>();
+  bool hasAccepted = false;
+
+  late final StreamSubscription<SessionProposalEvent?> listener;
+  @override
+  void initState() {
+    super.initState();
+
+    listener = wcService.onProposalExpire.stream.listen((event) {
+      if (event!.id == widget.pairingId) {
+        Navigator.of(context).pop(false);
+      }
+    });
+  }
+
+  @override
+  void dispose() {
+    listener.cancel();
+    super.dispose();
+  }
+
+  // Gets the text for WC method available from connection
+  Widget getMethod(String? text, {bool isGranted = true}) {
+    return Row(
+      children: [
+        Image.asset(isGranted
+            ? "assets/images/permission-granted.png"
+            : "assets/images/permission-denied.png"),
+        ThemedControls.spacerHorizontalSmall(),
+        Expanded(
+            child: Text(
+          text!,
+          style: TextStyles.walletConnectDapPermission,
+        ))
+      ],
+    );
+  }
+
+  // Gets a list of all requested WC methods
+  List<Widget> getMethods() {
+    final l10n = l10nOf(context);
+
+    List<Widget> methods = [];
+
+    for (var string in widget.pairingMethods) {
```

Comment:
please use the constants defined for wc method names

Reply:
@sallymoc Fixed and pushed to branch. Please check again and resolve